WebRTC Encoded Transforms docs (#28439)
mdn/content: files/en-us/web/api/rtctransformevent/index.md

At time of writing there is just one event based on `RTCTransformEvent`: {{domxref("DedicatedWorkerGlobalScope.rtctransform_event", "rtctransform")}}.

{{InheritanceDiagram}}

## Instance properties

_Since `RTCTransformEvent` is based on {{domxref("Event")}}, its properties are also available._

- {{domxref("RTCTransformEvent.transformer")}} {{ReadOnlyInline}}
  - : Returns the {{domxref("RTCRtpScriptTransformer")}} associated with the event.

## Transform event types

There is only one type of transform event.

### `rtctransform`

The {{domxref("DedicatedWorkerGlobalScope.rtctransform_event", "rtctransform")}} event is fired at the worker global scope on construction of an associated {{domxref("RTCRtpScriptTransform")}}, and whenever a new encoded video or audio frame is enqueued for processing.

You can add a `rtctransform` event listener to be notified when the new frame is available using either {{domxref("EventTarget.addEventListener", "DedicatedWorkerGlobalScope.addEventListener()")}} or the `onrtctransform` event handler property.

## Example

This example creates an event listener for the {{domxref("DedicatedWorkerGlobalScope.rtctransform_event", "rtctransform")}} event.

> The code first creates a {{domxref("TransformStream")}} (implementation not show), depending on the value of `options` passed from a {{domxref("RTCRtpScriptTransform")}} constructor in the main-thread.

Reviewer: Does this make sense? Maybe we can use the same above if so.

```suggestion
The example assumes we have a {{domxref("TransformStream")}} with an `options` object passed from a {{domxref("RTCRtpScriptTransform")}} constructor in the main-thread.
```
